frontend/src/container/AdvancedSearch.jsx

```diff
@@ -0,0 +1,303 @@
+
+    const fetchGetSearchedKeywords = await props.dispatch(getSearchedKeywords(
+      searchObject.contentTypes,
+      searchObject.searchedKeywords,
+      FIRST_PAGE,
+      (searchObject.numberResultsByPage * searchObject.currentPage),
+      searchObject.showArchived,
+      searchObject.showDeleted,
+      searchObject.showActive
+    ))
+
+    switch (fetchGetSearchedKeywords.status) {
+      case 200:
+        props.dispatch(setSearchedKeywords(searchObject.searchedKeywords))
+        props.dispatch(setSearchResultsList(fetchGetSearchedKeywords.json.contents))
+        props.dispatch(setCurrentNumberPage(searchObject.currentPage))
+        props.dispatch(setNumberResultsByPage(searchObject.numberResultsByPage))
+        this.setState({ totalHits: fetchGetSearchedKeywords.json.total_hits })
+        break
+      default:
+        props.dispatch(newFlashMessage(props.t('An error has happened'), 'warning'))
+        break
+    }
+  }
+
+  getContentName = (content) => {
+    // FIXME - GB - 2019-06-04 - we need to have a better way to check if it is a file than using contentType[1]
+    // https://github.com/tracim/tracim/issues/1840
+    const { props } = this
+    let contentName = ''
+
+    if (props.contentType.length > 1) {
+      contentName = content.contentType === props.contentType[1].slug ? content.filename : content.label
+    } else {
+      props.dispatch(newFlashMessage(props.t('An error has happened'), 'warning'))
+    }
+    return contentName
+  }
+
+  handleClickSeeMore = async () => {
+    const { props } = this
+    const NEXT_PAGE = props.searchResult.currentNumberPage + 1
+    const searchObject = parseSearchUrl(qs.parse(props.location.search))
+
+    const fetchGetSearchedKeywords = await props.dispatch(getSearchedKeywords(
+      searchObject.contentTypes,
+      props.searchResult.searchedKeywords,
+      NEXT_PAGE,
+      props.searchResult.numberResultsByPage,
+      searchObject.showArchived,
+      searchObject.showDeleted,
+      searchObject.showActive
+    ))
+
+    switch (fetchGetSearchedKeywords.status) {
+      case 200:
+        props.dispatch(setCurrentNumberPage(NEXT_PAGE))
+        props.dispatch(appendSearchResultsList(fetchGetSearchedKeywords.json.contents))
+        this.setState({ totalHits: fetchGetSearchedKeywords.json.total_hits })
+        props.history.push(PAGE.SEARCH_RESULT + '?' + qs.stringify({ ...qs.parse(props.location.search), p: NEXT_PAGE }, { encode: true }))
+        break
+      default:
+        props.dispatch(newFlashMessage(props.t('An error has happened'), 'warning'))
+        break
+    }
+  }
+
+  handleClickFilterMenu = () => this.setState(prev => ({ isFilterMenuOpen: !prev.isFilterMenuOpen }))
+
+  handleClickSearch = searchedKeywords => {
+    const { props } = this
+    const FIRST_PAGE = 1
+
+    props.dispatch(setSearchedKeywords(searchedKeywords))
+    props.dispatch(setCurrentNumberPage(FIRST_PAGE))
+    props.history.push(PAGE.SEARCH_RESULT + '?' + qs.stringify({
```

Comment:
This is the same issue with routing, clicking the search button should only do a history.push and the component should update redux on reaction to the new route

Reply:
this won't be "required changed", and we can talk about it a bit more if you need